### PR TITLE
HPCC-20195 Roxie MySQL cache options for ConfigManager 2

### PR DESCRIPTION
**Pull request by kenrowland**
Add MySQL cache options to new roxie.xsd

Signed-off-by: Ken Rowland <kenneth.rowland@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

File: initfiles/componentfiles/configschema/xsd/roxie.xsd [hunk 1]
```diff
@@ -188,6 +188,12 @@
             <xs:attribute name="leafCacheMem" type="xs:nonNegativeInteger" hpcc:displayName="Leaf Cache Memory Size (Mb)"  use="optional" default="50" hpcc:tooltip="Size (in Mb) of leaf index page cache"/>
             <xs:attribute name="nodeCachePreload" type="xs:boolean" hpcc:displayName="Enable Node Cache Preload"  use="optional" default="false" hpcc:tooltip="Prefill the node cache with all non-leaf pages from all indexes"/>
             <xs:attribute name="nodeCacheMem" type="xs:nonNegativeInteger" hpcc:displayName="Node Cache Memory Size (Mb)"  use="optional" default="100" hpcc:tooltip="Size (in Mb) of non-leaf index page cache"/>
+            <xs:attribute name="mysqlCacheCheckPeriod" type="xs:nonNegativeInteger" hpcc:displayName="MySQL Cache Check Period(ms)" use="optional" default="10000"
+                hpcc:tooltip="Time to wait (ms) between checking if any cached MySQL connections can be closed"/>
+            <xs:attribute name="mysqlCacheTimeoutPeriod" type="xs:nonNegativeInteger" hpcc:displayName="MySQL Cache Timeout Period(ms)" use="optional" default="60000"
```
Comment (rpastrana) [inline initfiles/componentfiles/configschema/xsd/roxie.xsd]: xs:nonNegativeInteger seems reasonable for these, I don't remember how they were declared in the current xsd, but I'd confirm that -1,etc don't have a special meaning.

File: initfiles/componentfiles/configschema/xsd/roxie.xsd [hunk 2]
```diff
@@ -188,6 +188,12 @@
             <xs:attribute name="leafCacheMem" type="xs:nonNegativeInteger" hpcc:displayName="Leaf Cache Memory Size (Mb)"  use="optional" default="50" hpcc:tooltip="Size (in Mb) of leaf index page cache"/>
             <xs:attribute name="nodeCachePreload" type="xs:boolean" hpcc:displayName="Enable Node Cache Preload"  use="optional" default="false" hpcc:tooltip="Prefill the node cache with all non-leaf pages from all indexes"/>
             <xs:attribute name="nodeCacheMem" type="xs:nonNegativeInteger" hpcc:displayName="Node Cache Memory Size (Mb)"  use="optional" default="100" hpcc:tooltip="Size (in Mb) of non-leaf index page cache"/>
+            <xs:attribute name="mysqlCacheCheckPeriod" type="xs:nonNegativeInteger" hpcc:displayName="MySQL Cache Check Period(ms)" use="optional" default="10000"
```
Comment (rpastrana) [inline initfiles/componentfiles/configschema/xsd/roxie.xsd]: I thought you were pushing to move away from the defaults in the config and label these as "defaultincode" ?

**Review by rpastrana — COMMENTED** [review]
@kenrowland a couple of comments/questions

File: initfiles/componentfiles/configschema/xsd/roxie.xsd (hunk 2 above)
Comment (kenrowland) [reply initfiles/componentfiles/configschema/xsd/roxie.xsd]: This PR is just to capture the attributes. The full changes for default support will be done as part of a full roxie xsd review.

File: initfiles/componentfiles/configschema/xsd/roxie.xsd (hunk 1 above)
Comment (kenrowland) [reply initfiles/componentfiles/configschema/xsd/roxie.xsd]: These are all declared exactly as added in the legacy xsd file.

**Review by kenrowland — COMMENTED** [review]
@rpastrana Please see responses.

**Review by rpastrana — APPROVED** [review]
@kenrowland approved based on your claims. Make sure they are correct.

**Comment by kenrowland** [comment]
@ghalliday Please merge (I think it is you doing this now)

**Pull request merged**